torch/nn/modules/transformer.py

```diff
@@ -55,26 +55,35 @@ def __init__(self, d_model=512, nhead=8, num_encoder_layers=6,
         self.d_model = d_model
         self.nhead = nhead
 
-    def forward(self, src, tgt, src_mask=None, tgt_mask=None, memory_mask=None):
+    def forward(self, src, tgt, src_mask=None, tgt_mask=None,
+                memory_mask=None, key_padding_mask=None):
```

Comment:
the key_padding masks for encoder and decoder are not necessary to be same, though they could be. So maybe it's better to have encoder_key_padding_mask and decoder_key_padding_mask and both are passed to the self-attention layer in encoder/decoder, respectively.

Reply:
I think you still want the option of masking the memory layer, as @sebamenabar pointed out

Reply:
So I have no explicit opposition on this. :)